src/software/ai/hl/stp/action/intercept_ball_action.cpp

```diff
@@ -73,15 +73,26 @@ void InterceptBallAction::calculateNextIntent(IntentCoroutine::push_type& yield)
         bool point_in_front_of_ball =
             pointInFrontVector(ball.position(), ball.velocity(), closest_point);
 
-        // We add 1e-6 to avoid division by 0 without affecting the result significantly
-        Duration ball_time_to_position = Duration::fromSeconds(
-            distance(closest_point, ball.position()) / (ball.velocity().length() + 1e-6));
+        Duration ball_time_to_position;
+        bool ball_not_moving = false;
+
+        if (ball.velocity().length() != 0)
+        {
+            ball_time_to_position = Duration::fromSeconds(
+                distance(closest_point, ball.position()) / (ball.velocity().length()));
+        }
+        else
+        {
+            ball_not_moving = true;
+        }
+
         Duration robot_time_to_pos = getTimeToPositionForRobot(
             *robot, closest_point, ROBOT_MAX_SPEED_METERS_PER_SECOND,
             ROBOT_MAX_ACCELERATION_METERS_PER_SECOND_SQUARED);
 
         std::optional<Point> intercept_pos = std::nullopt;
-        if (point_in_front_of_ball && (ball_time_to_position > robot_time_to_pos))
+        if (ball_not_moving ||
+            (point_in_front_of_ball && (ball_time_to_position > robot_time_to_pos)))
```

Comment:
I think this should be point_in_front_of_ball && (ball_not_moving ||ball_time_to_position > robot_time_to_pos)

Reply:
Done.